doc/source/ray-debugging.rst

```diff
@@ -0,0 +1,174 @@
+Ray Debugging
+=============
+
+Ray has a built in debugger that allows you to debug your distributed applications.
+
+.. note::
+
+    It is currently an experimental feature and under active development. Interfaces are subject to change.
+
+Getting Started
+---------------
+
+Take the following example that is modified from the Ray README:
+
+.. code-block:: python
+
+    import ray
+    ray.init()
+
+    @ray.remote
+    def f(x):
+        ray.util.pdb.set_trace()
+        return x * x
+
+    futures = [f.remote(i) for i in range(2)]
+    print(ray.get(futures))
+
+Put the program into a file named ``debugging.py`` and execute it using:
+
+.. code-block:: bash
+
+    python debugging.py
+
+
+Each of the 4 executed tasks will drop into a breakpoint when the line
+``ray.util.pdb.set_trace()`` is executed. You can attach to the debugger by running
```

Comment:
Can you say where you can run this command? is it on your laptop, on the head node, on any node?

Reply:
oh yeah good point, fixed!